conda_pupa/convert_tree.py

```diff
@@ -35,6 +41,35 @@ def parse_libmamba_error(message: str):
             yield match.group(1)
 
 
+class ReloadingLibMambaSolver(LibMambaSolver):
+    """
+    Reload channels as we add newly converted packages.
+    LibMambaIndexHelper appears to be addressing C++ singletons or global state.
+    """
+
+    def _collect_all_metadata(
+        self,
+        channels: Iterable[Channel],
+        conda_build_channels: Iterable[Channel],
+        subdirs: Iterable[str],
+        in_state: SolverInputState,
+    ) -> LibMambaIndexHelper:
+        index = LibMambaIndexHelper(
+            channels=[*conda_build_channels, *channels],
+            subdirs=subdirs,
+            repodata_fn=self._repodata_fn,
+            installed_records=(
+                *in_state.installed.values(),
+                *in_state.virtual.values(),
+            ),
+            pkgs_dirs=context.pkgs_dirs if context.offline else (),
+        )
+        for channel in channels:
```

Comment:
```suggestion
        for channel in channels:
```
Only reload local channel...